.github/workflows/e2e-tests.yml

```
@@ -140,6 +140,8 @@ jobs:
       continue-on-error: true
       run: |
         pnpm run tunnel:down
+        # Update permissions to workaround https://github.com/actions/cache/issues/753
+        sudo chown -R runner:docker ../../../../../tools/docker
```

Might be slightly better to do like
```suggestion
        sudo chown -R runner:docker "$GITHUB_WORKSPACE/tools/docker"
```
instead of relying on `matrix.path` always needing `../../../../../` to get back there.

Also you might just `rm -rf` it instead of chowning, since that doesn't depend on the user being "runner:docker". Is there anything under there that needs to be kept?
```suggestion
        sudo rm -rf "$GITHUB_WORKSPACE/tools/docker"
```